plugin/storage/es/factory.go

```diff
@@ -58,26 +57,25 @@ type Factory struct {
 
 	newClientFn func(c *config.Configuration, logger *zap.Logger, metricsFactory metrics.Factory) (es.Client, error)
 
-	primaryConfig *config.Configuration
-	archiveConfig *config.Configuration
+	config *config.Configuration
 
-	primaryClient atomic.Pointer[es.Client]
-	archiveClient atomic.Pointer[es.Client]
+	client atomic.Pointer[es.Client]
 
 	watchers []*fswatcher.FSWatcher
 }
 
 // NewFactory creates a new Factory.
-func NewFactory() *Factory {
+func NewFactory(namespace string) *Factory {
 	return &Factory{
-		Options:     NewOptions(primaryNamespace, archiveNamespace),
+		Options:     NewOptions(namespace),
 		newClientFn: config.NewClient,
 		tracer:      otel.GetTracerProvider(),
 	}
 }
 
 func NewFactoryWithConfig(
 	cfg config.Configuration,
+	namespace string,
```

Comment:
doesn't make sense. This function is used by v2 where the configuration is already defined in the struct, whereas v1 needs a namespace string in order to form CLI flag names.

Reply:
The reason for this is because both v1 and v2 initialize the same way. I guess for the case of v2 - the `Options` struct doesn't really matter and we can just directly set the config to what's passed in. Would that be okay?

Reply:
yes.

Reply:
done